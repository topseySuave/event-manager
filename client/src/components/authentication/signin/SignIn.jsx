import React, { Component } from 'react';
import DocumentTitle from 'react-document-title';
import AuthHeader from '../AuthHeader';
import AuthFooter from '../AuthFooter';
import SignInForm from './SignInForm';

/**
   * SignIn Class Component
   * */
class SignIn extends Component {
  /**
   * render method
   * @returns { Component }
   * @memberof SignIn
   * */
  render() {
    return (
      <DocumentTitle title="Sign In | Boots Events Manager">
        <div>
          <AuthHeader />

          <div className="signin__card_holdr wow fadeInUp">
            <div className="container">
              <div className="row">
                <SignInForm />
              </div>
            </div>
          </div>

          <div className="empty" />

          <AuthFooter />
        </div>
      </DocumentTitle>
    );
  }
}

export default SignIn;
