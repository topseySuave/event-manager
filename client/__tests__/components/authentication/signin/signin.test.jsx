import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactRouterEnzymeContext from 'react-router-enzyme-context';
import SignIn from
  '../../../../src/components/authentication/signin/SignIn';

describe('Authentication Sign In Component', () => {
  const options = new ReactRouterEnzymeContext();
  let wrapper = shallow(<SignIn />);
  test('should make a snapshot of the SignIn component', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBe(5);
    expect(wrapper).toMatchSnapshot();
  });

  test('should always render the header and footer component', () => {
    const AuthHeader = wrapper.find('AuthHeader');
    expect(AuthHeader.length).toBe(1);
    const AuthFooter = wrapper.find('AuthFooter');
    expect(AuthFooter.length).toBe(1);
    const signinCardHolder = wrapper.find('.signin__card_holdr');
    expect(signinCardHolder.exists()).toBeTruthy();
  });
});
