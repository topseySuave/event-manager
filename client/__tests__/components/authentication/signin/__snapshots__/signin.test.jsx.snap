// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Sign In Component make a snapshot of the SignIn component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <AuthHeader />
        <div
          className="signin__card_holdr wow fadeInUp"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <Connect(SignInForm) />
            </div>
          </div>
        </div>
        <div
          className="empty"
        />
        <AuthFooter />
      </div>,
      "title": "Sign In | Boots Events Manager",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AuthHeader />,
          <div
            className="signin__card_holdr wow fadeInUp"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <Connect(SignInForm) />
              </div>
            </div>
          </div>,
          <div
            className="empty"
          />,
          <AuthFooter />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="row"
              >
                <Connect(SignInForm) />
              </div>
            </div>,
            "className": "signin__card_holdr wow fadeInUp",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <Connect(SignInForm) />
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(SignInForm) />,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "empty",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <AuthHeader />
          <div
            className="signin__card_holdr wow fadeInUp"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <Connect(SignInForm) />
              </div>
            </div>
          </div>
          <div
            className="empty"
          />
          <AuthFooter />
        </div>,
        "title": "Sign In | Boots Events Manager",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AuthHeader />,
            <div
              className="signin__card_holdr wow fadeInUp"
            >
              <div
                className="container"
              >
                <div
                  className="row"
                >
                  <Connect(SignInForm) />
                </div>
              </div>
            </div>,
            <div
              className="empty"
            />,
            <AuthFooter />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <div
                  className="row"
                >
                  <Connect(SignInForm) />
                </div>
              </div>,
              "className": "signin__card_holdr wow fadeInUp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row"
                >
                  <Connect(SignInForm) />
                </div>,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(SignInForm) />,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "empty",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
